### PR TITLE
feat: rewardee karma in redis db

### PR DESCRIPTION
**Pull request by dwight-schrute**

### CODE REVIEW

File: src/repository/Repository.ts [hunk 1]
```diff
@@ -1,21 +1,86 @@
...
+                user: userName,
+                karma: info.karma,
+                channel: guildChannels.get(info.channelID)?.name
+            }
+        });
+        return userKarmaList.sort((a, b) => b.karma - a.karma);
+    }
+
+    private setRewardeeChannelKarma(userID: string, channelID: string): void {
+        this.redisClient.GET(`karmaInfo:${userID}:${channelID}`, (err: Error, reply: string) => {
```
Comment (JimWithoutJam) [inline src/repository/Repository.ts:69]: we were planning to have a setup like `"karmaInfo:userId": { channelId1: karma, channelId2: karma }`. Any blockers with that approach?

File: src/repository/Repository.ts [hunk 2]
```diff
@@ -1,21 +1,86 @@
...
+            const [_, userID, channelID] = key.split(':');
+            return {
+                userID,
+                channelID,
+                karma: parseInt(karma)
+            }
+        }));
+        const userIDList = karmaInfo.map(info => info.userID);
+        
+        const guild = await client.guilds.fetch('842338736124854293');
```
Comment (JimWithoutJam) [inline src/repository/Repository.ts]: Let's extract this part to env variable?

File: src/repository/Repository.ts [hunk 3]
```diff
@@ -1,21 +1,86 @@
...
+        this.getKeysAsync = promisify(this.redisClient.keys).bind(this.redisClient);
     }
 
-    public async saveRewardEvent(rewarder: User, rewardees: User[], channelID: string) {
-        const eventInstance = new RewardEvent();
-        eventInstance.rewarder = rewarder.id;
-        eventInstance.timestamp = new Date();
-        eventInstance.karma = 1;
-        eventInstance.channel = channelID;
+    public saveRewardEvent(rewarder: User, rewardees: User[], channelID: string) {
```
Comment (JimWithoutJam) [inline src/repository/Repository.ts]: Our future plan includes having setup for `@mention +++`. That had mean rewarding multiple karmas. Can we start accepting an optional parameter `karma: number = 1`?

**Review by JimWithoutJam — CHANGES_REQUESTED** [review]

File: src/repository/Repository.ts (hunk 1 above)
Comment (dwight-schrute) [reply src/repository/Repository.ts:69]: No blockers with the approach, but I don't think that will improve the performance or anything. Let's go with the current approach, we will improve it if the need be.

Comment (JimWithoutJam) [reply src/repository/Repository.ts:69]: cool ok

**Pull request merged**